Agents/PropertyValueEstimationAgent/agent/agentlogic/value_estimation.py

```diff
@@ -0,0 +1,214 @@
+        Returns:
+            Graph to instantiate/update property market value
+        """
+
+        # Initialise market value and return triples
+        market_value = None
+        triples = ''
+
+        # Prio 1: Check if transaction record and property price index are provided
+        # (i.e. market value assessment based on previous transaction)
+        if transaction_iri and prop_price_index_iri:
+            # Initialise TS client
+            ts_client = TSClient(kg_client=self.sparql_client)
+            # 1) Retrieve representative UK House Price Index and parse as Series (i.e. unwrap Java data types)
+            # UKHPI was set at a base of 100 in January 2015, and reflects the change in value of residential property since then
+            # (https://landregistry.data.gov.uk/app/ukhpi/doc)
+            try:
+                # Retrieve time series in try-with-resources block to ensure closure of RDB connection
+                with ts_client.connect() as conn:
+                    ts = ts_client.tsclient.getTimeSeries([prop_price_index_iri], conn)
+                dates = [d.toString() for d in ts.getTimes()]
+                values = [v for v in ts.getValues(prop_price_index_iri)]
+            except Exception as ex:
+                self.logger.error('Error retrieving/unwrapping Property Price Index time series')
+                raise TSException('Error retrieving/unwrapping Property Price Index time series') from ex
+
+            # Create UKHPI series with conditioned date index
+            ukhpi = pd.Series(index=dates, data=values)
+            ukhpi = ukhpi.astype(float)
+            ukhpi.index = pd.to_datetime(ukhpi.index, format=TIME_FORMAT_LONG)
+            ukhpi.sort_index(ascending=False, inplace=True)
+            ukhpi.index = ukhpi.index.strftime(TIME_FORMAT_SHORT)
+
+            # 2) Retrieve previous sales transaction details for previous transaction IRI
+            #    and adjust to current market value
+            res = self.sparql_client.get_transaction_details(transaction_iri)
+            ukhpi_now = ukhpi.iloc[0]
+            ukhpi_old = ukhpi[res['date']]
+            market_value = res['price'] * ukhpi_now / ukhpi_old
+
+        # Prio 2: Otherwise assess market value based on FloorArea and AveragePricePerSqm
+        elif avgsqm_price_iri and floor_area_iri and not market_value:
+            # TODO: How to handle cases where avgsqm_price is just not calculated yet?
+            # --> Suggestion: Avoid this case by marking AvgSqmPrice as Synchronous Derivation initially
+            res = self.sparql_client.get_floor_area_and_avg_price(floor_area_iri)
+            market_value = res['floor_area'] * res['avg_price']
+
+        if market_value:
+            # Round property market value to full kGBP
+            market_value = round(market_value/1000)*1000
+            # Create instantiation/update triples
+            market_value_iri = KB + 'AmountOfMoney' + str(uuid.uuid4())
```

Comment:
This is nitpicking, but I would prefer to create `KB + 'AmountOfMoney_' + str(uuid.uuid4())` (with the underscore), like how you did with the `measure_iri` in `self.sparql_client.instantiate_property_value`.

Reply:
Of course you do. So do I ;-) Just missed it, but addressed in [1447df6](https://github.com/cambridge-cares/TheWorldAvatar/pull/454/commits/1447df66cc776b90a98c00c2abfd1fe42889e786)